client/blocks/app-banner/index.jsx

```diff
@@ -228,13 +233,15 @@ export function buildDeepLinkFragment( currentRoute, currentSection ) {
 const mapStateToProps = state => {
 	const sectionName = getSectionName( state );
 	const isNotesOpen = isNotificationsOpen( state );
+	const isInActivityLog = window.location.pathname.indexOf( '/stats/activity/' ) !== -1;
```

Comment:
As a general note, please check for `window` before using it. Some pages are SSR'd and we can't assume that the env is a browser. 

Since we already have a few utility function to separate out this logic, let's move this check to `getCurrentSection` in `client/blocks/app-banner/utils` where we can return null for this case instead. Current route is also available from global state, so let's try:
```jsx
export function getCurrentSection( currentSection, { isNotesOpen, currentRoute } ) {
	if ( isNotesOpen ) {
		return NOTES;
	}

    if ( currentRoute && currentRoute.indexOf( '/stats/activity/' ) !== -1 ) {
        //don't show app banner in activity log
        return null;
    }

	if ( includes( ALLOWED_SECTIONS, currentSection ) ) {
		return currentSection;
	}

	return null;
}
```

And in `mapStateToProps` this changes to:
```jsx
const mapStateToProps = state => {
	const sectionName = getSectionName( state );
	const isNotesOpen = isNotificationsOpen( state );
    const currentRoute = getCurrentRoute( state );

	return {
		dismissedUntil: getPreference( state, APP_BANNER_DISMISS_TIMES_PREFERENCE ),
		currentSection: getCurrentSection( sectionName, { isNotesOpen, currentRoute } ),
		currentRoute,
		fetchingPreferences: isFetchingPreferences( state ),
		siteId: getSelectedSiteId( state ),
	};
};
```

And we can remove any other checks for `isInActivityLog`. You're welcome to add unit test for `getCurrentSection` in utils.js. Ping me if you need help with that.